test/browser/newsletter-signup.test.jsx

```diff
@@ -0,0 +1,65 @@
+var NewsletterSignupForm = require('../../components/newsletter-signup/SignupForm.jsx');
+var validateSignupForm = require('../../components/newsletter-signup/validateSignupForm.js');
+
+var stubContext = require('./stub-context.jsx');
+var Util = require('../util.js');
+
+var GOOD_EMAIL_EXAMPLES = [
+  "hello@example.com",
+  "hello123@example.com",
+  "hello+test@example.com",
+  "hello123+test@example.com",
+  "hello!@example.com",
+  "!@example.com",
+  "#!$%&'*+-/=?^_`{}|~@example.org"
+];
+
+var BAD_EMAIL_EXAMPLES = [
+  "",
+  "h.e.llo",
+  "h@e@llo@example.com",
+  "@example.com",
+  "@example.com.",
+  "h..ello@examplecom",
+  "hello@example..com",
+  "hel lo@example.com",
+  "@.com",
+  "@",
+  "hello",
+  ".com",
+  "."
+];
```

Comment:
Is this helpful?
https://help.salesforce.com/HTViewSolution?id=000001145&language=en_US

On Mon, Jun 27, 2016 at 4:53 PM, Mavis Ou <notifications@github.com> wrote:

> In test/browser/newsletter-signup.test.jsx
> <https://github.com/mozilla/learning.mozilla.org/pull/1911#discussion_r68671007>
> :
>
> > +
> > +var BAD_EMAIL_EXAMPLES = [
> > +  "",
> > +  "h.e.llo",
> > +  "h@e@llo@example.com",
> > +  "@example.com",
> > +  "@example.com.",
> > +  "h..ello@examplecom",
> > +  "hello@example..com",
> > +  "hel lo@example.com",
> > +  "@.com",
> > +  "@",
> > +  "hello",
> > +  ".com",
> > +  "."
> > +];
>
> @stephaniemcv <https://github.com/stephaniemcv> Do you by any chance know
> what validation rules Basket/Salesforce uses? GOOD_EMAIL_EXAMPLES and
> BAD_EMAIL_EXAMPLES are 2 lists of emails that I created for testing
> purposes but I have no idea if they match the specs on the backend 👀
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/learning.mozilla.org/pull/1911/files/71701e2a48b552ca4845927f91a176515f75d6ed#r68671007>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALQ9sKu8tJePA06-JNXTiGncBhccCTfIks5qQFRogaJpZM4I_kcz>
> .
>